pkg/controller/direct/metastore/backup_controller.go

```diff
@@ -0,0 +1,230 @@
+	id        *krm.BackupIdentity
+	desired   *krm.MetastoreBackup
+	actual    *pb.Backup
+	reader    client.Reader
+}
+
+var _ directbase.Adapter = &MetastoreBackupAdapter{}
+
+// Find retrieves the GCP resource.
+// Return true means the object is found. This triggers Adapter `Update` call.
+// Return false means the object is not found. This triggers Adapter `Create` call.
+// Return a non-nil error requeues the requests.
+func (a *MetastoreBackupAdapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("getting MetastoreBackup", "name", a.id)
+
+	req := &pb.GetBackupRequest{Name: a.id.String()}
+	metastorebackuppb, err := a.gcpClient.GetBackup(ctx, req)
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting MetastoreBackup %q: %w", a.id, err)
+	}
+
+	a.actual = metastorebackuppb
+	return true, nil
+}
+
+func (a *MetastoreBackupAdapter) resolveReferences(ctx context.Context) error {
+	obj := a.desired
+
+	if _, err := obj.Spec.ServiceRef.NormalizedExternal(ctx, a.reader, obj.Namespace); err != nil {
+		return fmt.Errorf("resolving serviceRef: %w", err)
+	}
+	return nil
+}
+
+// Create creates the resource in GCP based on `spec` and update the Config Connector object `status` based on the GCP response.
+func (a *MetastoreBackupAdapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("creating MetastoreBackup", "name", a.id)
+
+	if err := a.resolveReferences(ctx); err != nil {
+		return err
+	}
+
+	mapCtx := &direct.MapContext{}
+	desired := a.desired.DeepCopy()
+	resource := MetastoreBackupSpec_ToProto(mapCtx, &desired.Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	req := &pb.CreateBackupRequest{
+		Parent:   a.id.Parent().String(),
+		BackupId: a.id.ID(),
+		Backup:   resource,
+	}
+	op, err := a.gcpClient.CreateBackup(ctx, req)
+	if err != nil {
+		return fmt.Errorf("creating MetastoreBackup %s: %w", a.id, err)
+	}
+	created, err := op.Wait(ctx)
+	if err != nil {
+		return fmt.Errorf("MetastoreBackup %s waiting creation: %w", a.id, err)
+	}
+	log.V(2).Info("successfully created MetastoreBackup", "name", a.id)
+
+	status := &krm.MetastoreBackupStatus{}
+	status.ObservedState = MetastoreBackupObservedState_FromProto(mapCtx, created)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+	status.ExternalRef = direct.LazyPtr(a.id.String())
+	return createOp.UpdateStatus(ctx, status, nil)
+}
+
+// Update updates the resource in GCP based on `spec` and update the Config Connector object `status` based on the GCP response.
+// MetastoreBackup resource has no mutable fields as per the proto definition, so this function only updates the status.
+func (a *MetastoreBackupAdapter) Update(ctx context.Context, updateOp *directbase.UpdateOperation) error {
+	log := klog.FromContext(ctx)
+	log.V(2).Info("resource has no mutable fields, skipping update", "name", a.id)
```

Comment:
We still want to update KRM status in case of KCC acquiring the GCP resource.

Reply:
Done.